### PR TITLE
Plugable Dashboard.

### PR DESCRIPTION
**Pull request by abenari**
Added an interface for plugins to add widgets to the dashboard.
The user can customize the dashboard by adding removing and rearranging widgets. 
The customized dashboard can be saved locally in the browser local store (similar to cookie).

### CODE REVIEW

File: app/assets/javascripts/dashboard.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    widget.attr('data-hide', 'true').hide();
+    gridster.remove_widget(widget);
+    $(".gridster>ul").append(widget);
+    fill_restore_list();
+
+}
+
+function save_position(){
+    var positions = JSON.stringify(serialize_grid());
+    console.log(positions);
```
Comment (ohadlevy) [inline app/assets/javascripts/dashboard.js]: leftover or harmless?

**Comment by ohadlevy** [comment]
@abenari note the failing tests. Thanks!

**Comment by abenari** [comment]
rpms build is failing because of the new gem
I have created a separate branch for the jquery time-ago patch

**Comment by domcleal** [comment]
@abenari please try applying http://paste.fedoraproject.org/103544/6143121/raw/

File: app/assets/javascripts/dashboard.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+function fill_restore_list(){
+   $("ul>li.widget-restore").remove();
+   var restore_list = [];
+   var hidden_widgets = $(".gridster>ul>li[data-hide='true']");
+   if (hidden_widgets.exists()){
+       hidden_widgets.each(function(i, widget) {
+           var widget_name = $(widget).attr('data-name');
+           restore_list.push( "<li class='widget-restore'><a href='#' onclick='show_widget(\""+widget_name+"\")'>" + $.humanize(widget_name.replace('_widget', '')) + "</a></li>" );
```
Comment (domcleal) [inline app/assets/javascripts/dashboard.js]: humanize is a bad idea for i18n support, we should support proper names which can be translated.  Note that we have i18n functions in Javascript (double underscore) so it's possible to put the name in a Ruby file, have it extracted and then translate at the point of use in JS.

**Comment by ohadlevy** [comment]
merged as db6d37b for Foreman 1.6 thanks @abenari!

**Pull request closed without merging**